frack/app/models/user.py

```diff
@@ -1,17 +1,46 @@
-from sqlalchemy import Column, Integer, String
+from sqlalchemy import Column, Integer, String, Boolean, ForeignKey
+from sqlalchemy.orm import relationship
 from app import app, db
 import hashlib
 from os import urandom
 
 from itsdangerous import (TimedJSONWebSignatureSerializer
                           as Serializer, BadSignature, SignatureExpired)
 
+class UserType(db.Model):
+    __tablename__ = "usertype"
+    id = Column(Integer, primary_key=True)
+    name = Column(String(64), unique=True)
+
+    users = relationship("User", back_populates="user_type")
+
+class N0lleGroup(db.Model):
```

Comment:
Vill bara dubbellkolla att jag förstår den här strukturen rätt: vi har ett table som bara innehåller n0llegruppsnamn, och i användar-tablet har vi sedan en foreign key som är n0llegruppens ID. Utöver det så har varje användare också en typ, som antingen är phösargrupp eller n0llan.  Vad tjänar vi på detta jämfört med att ha n0llegrupp som en kolumn i användar-tablet? Känns som att den här strukturen leder till att vi måste hålla på med joins och så.

Reply:
Ja, om man gör på detta vis blir det mycket lättare att hantera nØllegrupper och användartyper. Förra året exempelvis hade vi skrivit fel namn på en av nØllegrupperna när vi la in användarna i databasen, så det blev lite småjobbigt att byta namn på nØllegruppen eftersom att man var tvungen att ändra alla användare som tillhörde den. Med denna lösning behöver man bara ändra namnet på nØllegruppen i databasen direkt!

Det blir lite wonky när det bara är en sträng som styr vilken typ en användare är, det kan bli jobbiga buggar som blir svåra att hitta ifall man råkar ha fel sträng. Det är lite som skillnaden mellan Javascript och C++, där Javascript inte klagar ifall man använder något som inte finns, medan C++ vägrar kompilera i så fall.

Reply:
Då kör vi på det, och får helt enkelt joina tables när det är dags att fetcha folk beroende på n0llegrupp